### ISSUE TITLE
After detailer goes so slow

### ISSUE THREAD

**Issue by tomboymad** [BeTheRobotSD/sdxl-comfyui-ultimate-workflow#2]
i think there is a bug with after detailer. If i do the following flow: txt2img > img2img > afterdetailer, it goes super super slow, but if i add ultimate after img, it goes fine... when i say slow i say 165sec for a it... (i'm using pose controlnet)

**Comment by BeTheRobotSD** [comment]
@tomboymad Are you using the Pose ControlNet on txt2img, img2img, and/or AfterDetailer?

**Comment by BeTheRobotSD** [comment]
In testing, I am seeing consistent times with both `img2img -> AfterDetailer @ 2.04s/it` and `Ultimate SD Upscale -> AfterDetailer @ 1.99s/it`. I am not using a ControlNet in any part of the flow during this test.

**Comment by tomboymad** [comment]
yes i was using controlnet pose when the thing goes slow... once i turned off, everything was ok..

**Comment by tomboymad** [comment]
> @tomboymad Are you using the Pose ControlNet on txt2img, img2img, and/or AfterDetailer?


yes i was using controlnet pose when the thing goes slow... once i turned off, everything was ok..

**Comment by BeTheRobotSD** [comment]
@tomboymad Just tested with and without ControlNet. I found that with ControlNet enabled, there was a 50% time increase going from `2s/it -> 3s/it @ 195s total`. This is expected as we are adding an unofficial ControlNet model into the mix. I also tested `img2img -> Ultimate SD Upscaler`, which ran at `2s/it` (without ControlNet) for a total time of 285 seconds. To me, this seems within normal ranges for generation given that I am running a 3090 FE. I will be closing this as there does not appear to be an issue with the workflow itself.
